quarkchain/cluster/shard_state.py

```diff
@@ -1413,11 +1419,7 @@ def add_root_block(self, root_block: RootBlock):
             )
 
         if self.header_tip != orig_header_tip:
-            header_tip_hash = self.header_tip.get_hash()
```

Comment:
so in the old code, we didn't update the evm state correctly after adding the root block？

Reply:
That is correct, and the added test will capture that.